api/apps/api/src/modules/geo-features/geo-feature-set.service.ts

```diff
@@ -115,33 +116,43 @@ export class GeoFeatureSetService {
     scenarioId: string,
     featureSpecification: SpecForGeofeature[],
   ): Promise<
-    ({
+    {
       scenarioId: string;
       featuresDataId: string;
-      fpf: number;
-      prop: number | undefined;
-    } & ScenarioFeaturesData)[]
+      fpf?: number | undefined;
+      prop?: number | undefined;
+    }[][]
   > {
     const repository = transactionalEntityManager.getRepository(
       ScenarioFeaturesData,
     );
-    return Promise.all(
+    const scenarioFeaturesData = Promise.all(
       featureSpecification
         .filter(
           (feature): feature is SpecForPlainGeoFeature =>
             feature.kind === 'plain',
         )
-        .map((feature) => {
-          return repository.save({
-            scenarioId,
-            featuresDataId: feature.featureId,
-            fpf:
-              feature.marxanSettings?.fpf ??
-              MarxanFeaturesMetadata.defaults.fpf,
-            prop: feature.marxanSettings?.prop,
-          });
+        .map(async (feature) => {
+          const featuresDataRepo = transactionalEntityManager.getRepository(
+            RemoteFeaturesData,
+          );
+          const featuresData = await featuresDataRepo
+            .find({ featureId: feature.featureId })
+            .then((result) =>
+              result.map((fd) => ({
+                scenarioId,
+                featuresDataId: fd.id,
+                fpf:
+                  feature.marxanSettings?.fpf ??
```

Comment:
Seems like a good candidate to be a default on entity/column instead scattering this logic within code. As anyone can just grab repo and add rows there, the default within service code could be overlooked.

Reply:
it does, though at this stage I'm not sure we'll keep any defaults here (fpf *may* be undefined) - I'd revisit this later.